Gateway fixes

shubham3121 opened the pull request:
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

- [x] Fix: Not able to login into Gateway as GUEST
- [x] Fix: Node peers not visible in case applied to gateway using the GuestClient
- [x] Fix: Double Wrapping of response in case of forwarded message
- [x] Fix: Message forwarding not working correctly for non `api_call` routes
     - APIs like login, register, metadata, not correctly forwarding API calls
     - [ ] Create a forward endpoint to handle all types of forwarding requests

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

shubham3121 committed:
fix unboxing of result in case of proxy request

shubham3121 committed:
make get_all_peers accessible to GUEST
fetch peers using node verify key

shubham3121 commented:
Changes from PR https://github.com/OpenMined/PySyft/pull/7888 will be pulled into this one.

iamtrask commented:
Hey @shubham3121 for my visibility could you expand upon this README so I know what's up here.

shubham3121 commented:
@iamtrask updated the readme

shubham3121 committed:
Update forward_message method to handle different routes
create a method to forward message if request is being send via proxy client
integrate forward message method to different connection routes

Co-authored-by: rasswanth <43314053+rasswanth-s@users.noreply.github.com>

shubham3121 committed:
add a notebook

shubham3121 committed:
Merge branch 'dev' into gateway-fixes

rasswanth-s commented on packages/syft/src/syft/client/client.py:
```diff
@@ -70,6 +73,34 @@ def upgrade_tls(url: GridURL, response: Response) -> GridURL:
     return url
 
 
+def forward_message_to_proxy(
```
Good one, shubham, for the generic structure, the code looks really good.

rasswanth-s committed:
fixe register flow in docker
replace routes exchanged with new SyftSuccess message
Refactored proxy_to function to return correct client type

rasswanth-s reviewed (approved):
Wonderful Work @shubham3121  🙌 
Kudos , for refactoring the forwarding endpoints in a nice generic way.

rasswanth-s committed:
added node type to worker settings

rasswanth-s committed:
added network node testing to enclave notebook

rasswanth-s committed:
Added dev mode to noteboks tests for enclave

shubham3121 committed:
add test for gateway

Co-authored-by: rasswanth <43314053+rasswanth-s@users.noreply.github.com>
